src/Orc.CsvHelper.Tests/Orc.CsvHelper.Tests.Shared/CsvExtensionsTest.cs

```diff
@@ -26,56 +26,68 @@ public class CsvExtensionsTest
-
-            var csvReaderService = serviceLocator.ResolveType<ICsvReaderService>();
-            var result = csvReaderService.ReadCsv<Operation, OperationCsvMap>(csvFilePath);
-
-            var expectedEnumerator = result.GetEnumerator();
-            foreach (var operation in result)
+            using (var tempFilesContext = new TemporaryFilesContext("CsvWriterService"))
             {
-                expectedEnumerator.MoveNext();
-                var expectedOperation = expectedEnumerator.Current;
+                var csvFilePath = tempFilesContext.GetFile("CsvWriterServiceTest.csv");
+                var originalData = CreateSampleOperations(operationCount).ToList();
+                await csvWriterService.WriteCsvAsync(originalData, csvFilePath);
 
-                Assert.AreEqual(operation.Id, expectedOperation.Id);
-                Assert.AreEqual(operation.Name, expectedOperation.Name);
-                Assert.AreEqual(operation.StartTime, expectedOperation.StartTime);
-                Assert.AreEqual(operation.Duration, expectedOperation.Duration);
-                Assert.AreEqual(operation.Quantity, expectedOperation.Quantity);
-                Assert.AreEqual(operation.Enabled, expectedOperation.Enabled);
-            }
+                var result = csvReaderService.ReadCsv<Operation, OperationCsvMap>(csvFilePath)
+                    .ToList();
 
-            if (File.Exists(csvFilePath))
-            {
-                File.Delete(csvFilePath);
+                Assert.AreEqual(operationCount, result.Count);
+                using (var expectedEnumerator = originalData.GetEnumerator())
+                {
+                    foreach (var operation in result)
+                    {
+                        expectedEnumerator.MoveNext();
+                        var expectedOperation = expectedEnumerator.Current;
+
+                        Assert.AreEqual(operation.Id, expectedOperation.Id);
+                        Assert.AreEqual(operation.Name, expectedOperation.Name);
+                        Assert.AreEqual(operation.StartTime, expectedOperation.StartTime);
+                        Assert.AreEqual(operation.Duration, expectedOperation.Duration);
+                        Assert.AreEqual(operation.Quantity, expectedOperation.Quantity);
+                        Assert.AreEqual(operation.Enabled, expectedOperation.Enabled);
+                    }
+                }
             }
         }
 
         [Test]
         public void FromCsvFile()
         {
             var serviceLocator = ServiceLocator.Default;
-            var csvReaderService = serviceLocator.ResolveType<ICsvReaderService>();
+            var typeFactory = serviceLocator.ResolveType<ITypeFactory>();
+            var csvReaderService = typeFactory.CreateInstanceWithParametersAndAutoCompletion<CsvReaderService>();
+            var csvFilePath = Path.Combine(AssemblyDirectoryHelper.GetCurrentDirectory(), @"TestData\Operations.csv");
```

Comment:
don't get: this path: Path.Combine(AssemblyDirectoryHelper.GetCurrentDirectory(), @"TestData\Operations.csv"); it is already in csvFilePath